pkg/virtctl/expose/expose_test.go

```diff
@@ -109,325 +96,256 @@ var _ = Describe("Expose", func() {
+			It("should succeed", func() {
+				cmd := clientcmd.NewRepeatableVirtctlCommand(expose.COMMAND_EXPOSE, "vmi", vmName, "--name", "my-service",
+					"--port", "9999")
+				Expect(cmd()).To(Succeed())
+			})
+		})
+		Context("With cluster-ip on a vm that has no label", func() {
+			It("should fail", func() {
+				cmd := clientcmd.NewRepeatableVirtctlCommand(expose.COMMAND_EXPOSE, "vmi", vmNoLabelName, "--name", "my-service",
+					"--port", "9999")
+				Expect(cmd()).NotTo(Succeed())
+			})
+		})
+		Context("With cluster-ip on an unknown vm", func() {
+			It("should fail on a vmi", func() {
+				cmd := clientcmd.NewRepeatableVirtctlCommand(expose.COMMAND_EXPOSE, "vmi", unknownVM, "--name", "my-service",
+					"--port", "9999")
+				Expect(cmd()).NotTo(Succeed())
+			})
+			It("should fail on a vm", func() {
+				cmd := clientcmd.NewRepeatableVirtctlCommand(expose.COMMAND_EXPOSE, "vm", unknownVM, "--name", "my-service",
+					"--port", "9999")
+				Expect(cmd()).NotTo(Succeed())
+			})
+		})
+		Context("With node-port service on a vm", func() {
+			It("should succeed", func() {
+				cmd := clientcmd.NewRepeatableVirtctlCommand(expose.COMMAND_EXPOSE, "vmi", vmName, "--name", "my-service",
+					"--port", "9999", "--type", "NodePort")
+				Expect(cmd()).To(Succeed())
+			})
+		})
+		Context("With cluster-ip on an vm", func() {
+			It("should succeed", func() {
+				err := clientcmd.NewRepeatableVirtctlCommand(expose.COMMAND_EXPOSE, "vm", vmName, "--name", "my-service",
+					"--port", "9999")
+				Expect(err()).To(Succeed())
+			})
+		})
+		Context("With cluster-ip on an vm replica set", func() {
+			It("should succeed", func() {
+				err := clientcmd.NewRepeatableVirtctlCommand(expose.COMMAND_EXPOSE, "vmirs", vmName, "--name", "my-service",
+					"--port", "9999")
+				Expect(err()).To(Succeed())
+			})
+		})
+		Context("With cluster-ip on an unknown vm replica set", func() {
+			It("should fail", func() {
+				cmd := clientcmd.NewRepeatableVirtctlCommand(expose.COMMAND_EXPOSE, "vmirs", unknownVM, "--name", "my-service",
+					"--port", "9999")
+				Expect(cmd()).NotTo(Succeed())
+			})
+		})
+		Context("With string target-port", func() {
+			It("should succeed", func() {
+				err := clientcmd.NewRepeatableVirtctlCommand(expose.COMMAND_EXPOSE, "vmi", vmName, "--name", "my-service",
+					"--port", "9999", "--target-port", "http")
+				Expect(err()).To(Succeed())
+			})
+		})
+		Context("With parametrized IPFamily", func() {
+			var dualStack = k8sv1.IPFamilyPolicyPreferDualStack
 
-				})
+			It("should succeed with IPv4", func() {
+				cmd := clientcmd.NewRepeatableVirtctlCommand(expose.COMMAND_EXPOSE, "vmi", vmName, "--name", "my-service",
+					"--port", "9999", "--target-port", "http", "--ip-family", "ipv4")
+				Expect(cmd()).To(Succeed(), "should succeed on an valid IP family - ipv4 is valid")
+				Expect(obtainedService).ToNot(BeNil())
+				Expect(obtainedService.Spec.IPFamilies).To(ConsistOf(k8sv1.IPv4Protocol))
```

Comment:
Any chance you can expose locally where this `obtainedService` is coming from?
It is very odd.